packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -41,6 +41,57 @@ class RenderableTiledMap {
       ? batchesByLayer.first
       : <String, SpriteBatch>{};
 
+  /// Changes the visibility of the corresponding layer, if different
+  void setLayerVisibility(int layerId, bool visibility) {
+    if (map.layers[layerId].visible != visibility) {
+      map.layers[layerId].visible = visibility;
+      refreshCache();
+    }
+  }
+
+  /// Gets the visibility of the corresponding layer
+  bool getLayerVisibility(int layerId) {
+    return map.layers[layerId].visible;
+  }
+
+  /// Changes the Gid of the corresponding layer at the given position,
+  /// if different
+  void setTileData({
+    required int layerId,
+    required int x,
+    required int y,
+    required Gid gid,
+  }) {
+    final layer = map.layers[layerId];
+    if (layer is TileLayer) {
+      final td = layer.tileData;
+      if (td != null) {
+        if (td[y][x].tile != gid.tile ||
+            td[y][x].flips.horizontally != gid.flips.horizontally ||
+            td[y][x].flips.vertically != gid.flips.vertically ||
+            td[y][x].flips.diagonally != gid.flips.diagonally) {
+          td[y][x] = gid;
+          refreshCache();
+        }
+      }
+    }
+  }
+
+  /// Gets the Gid  of the corresponding layer at the given position
+  Gid? getTileData({required int layerId, required int x, required int y}) {
+    final layer = map.layers[layerId];
+    if (layer is TileLayer) {
+      final td = layer.tileData;
+      if (td != null) {
+        return td[y][x];
+      } else {
+        return td?[y][x];
+      }
```

Comment:
```suggestion
      return layer.tileData?[y][x];
```

Reply:
Here, `layer` ist not of type `TileLayer`, so it has no member `tileData`.

Reply:
Yes it is? You do the check in the `if` outside of this change.

Reply:
```dart

final layer = map.layers[layerId];
if (layer is TileLayer) {
  final td = layer.tileData;
  if (td != null) {
    return td[y][x];
  } else {
    return td?[y][x];
  }
  return layer.tileData?[y][x]; // Dead Code Here
}  
return layer.tileData?[y][x]; // layer not of type TileLayer here  
```

Reply:
That line is something you accidentally added when you didn't commit my full suggestion the first time around. ;)
```dart
final layer = map.layers[layerId];
if (layer is TileLayer) {
  return layer.tileData?[y][x];
}  
return null;
```

Reply:
Great! Should I change this or should I press the botton above?

Reply:
The change above will be wrong still since it has the line that you mentioned, so probably easiest to just add those lines manually.

Reply:
Added a new suggestion here that you can just directly press commit on if you want:
https://github.com/flame-engine/flame/pull/1324/files#r787768163